lib/devise/controllers/helpers.rb

```diff
@@ -119,7 +119,7 @@ def authenticate_#{mapping}!(opts={})
           end
 
           def #{mapping}_signed_in?
-            !!current_#{mapping}
+            !!(@current_#{mapping} || warden.authenticated?(scope: :#{mapping}))
```

Comment:
The `current_#{mapping}` method still needs to call the `warden.authenticate` method, so if we keep calling it here we will still sign in the user if there are valid params. 

Should this method only call `warden.authenticated?(scope: :#{mapping})` or am I missing something?

Reply:
I think that's correct, we should probably change `#current_#{mapping}` too.

Now I think it'd be good to have a test for that - ensure we're not signing the user in when these methods are called - too.

Reply:
We are not actually calling `#current_#{mapping}`, but simply check whether the ivar `@current_#{mapping}` is set, so `warden.authenticate` won't be invoked.
I think that the decision of whether or not `#current_#{mapping}` should run authentication as well, falls outside the scope of this PR, since it explicitly concerns query methods.
Thanks for your reviews, I will take care of the tests!